managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2626,6 +2626,7 @@ void internalTrimLedgers(boolean isTruncate, CompletableFuture<?> promise) {
             }
 
             long slowestReaderLedgerId = -1;
+            long slowestNonDurationLedgerId = getTheSlowestNonDurationReadPosition().getLedgerId();
```

Comment:
I think this variable is essential. Do you think we can make it final?

Reply:
Done